### PR TITLE
Generic Content Type plugin

### PR DESCRIPTION
**Pull request by mullermp**
Move rest/request plugin to generic. Will add to all services

### CODE REVIEW

File: gems/aws-sdk-core/lib/aws-sdk-core/rest/request/content_type.rb [hunk 1]
```diff
@@ -1,51 +0,0 @@
-# frozen_string_literal: true
-
-module Aws
-  module Rest
-    # NOTE: headers could be already populated if specified on input shape
-    class ContentTypeHandler < Seahorse::Client::Handler
```
Comment (alextwoods) [inline gems/aws-sdk-core/lib/aws-sdk-core/rest/request/content_type.rb:6]: I don't think we can remove this.  This will break old service w/ new core

Comment (alextwoods) [reply gems/aws-sdk-core/lib/aws-sdk-core/rest/request/content_type.rb:6]: Edit: Actually nvm, this isn't dependend on by service clients directly, but through the protocol plugin.  So we may be okay.

File: gems/aws-sdk-core/lib/seahorse/client/plugins/content_type.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          private
+
+          def non_empty_body?(body)
+            body.respond_to?(:size) && body.size.positive?
+          end
+
+          # content-type defaults as noted here:
+          # rest-json: https://smithy.io/2.0/aws/protocols/aws-restxml-protocol.html#content-type
+          # rest-xml: https://smithy.io/2.0/aws/protocols/aws-restxml-protocol.html#content-type
+          def apply_default_content_type(context)
```
Comment (alextwoods) [inline gems/aws-sdk-core/lib/seahorse/client/plugins/content_type.rb:40]: I don't think this belongs in Seahorse - this is AWS specific logic per protocol.  I'd lean instead towards a generic base class which the protocols can extend and use.

File: gems/aws-sdk-core/lib/aws-sdk-core/rest/request/content_type.rb (hunk 1 above)
Comment (mullermp) [reply gems/aws-sdk-core/lib/aws-sdk-core/rest/request/content_type.rb:6]: It hasn't released yet. This is new code too

File: build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/plugin_list.rb [hunk 3]
```diff
@@ -40,6 +40,7 @@ def compute_plugins(options)
     def default_plugins
       {
         'Seahorse::Client::Plugins::ContentLength' => "#{seahorse_plugins}/content_length.rb",
+        'Seahorse::Client::Plugins::ContentType' => "#{seahorse_plugins}/content_type.rb",
```
Comment (jterapin) [inline build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/plugin_list.rb:43]: With this being default, should we relocate the setting of `content-type` for other protocols like [json](https://github.com/aws/aws-sdk-ruby/blob/09ee4c3e0b96e807d9fd1b9331c4387746f0896f/gems/aws-sdk-core/lib/aws-sdk-core/json/handler.rb#L23) here?

**Pull request closed without merging**